src/GafferScene/Instancer.cpp

```diff
@@ -766,11 +1255,49 @@ void Instancer::hash( const Gaffer::ValuePlug *output, const Gaffer::Context *co
 		scalePlug()->hash( h );
 		attributesPlug()->hash( h );
 		attributePrefixPlug()->hash( h );
+
+		seedEnablePlug()->hash( h );
+		seedVariablePlug()->hash( h );
+		seedPossibilitiesPlug()->hash( h );
+		seedPermutationPlug()->hash( h );
+		rawSeedPlug()->hash( h );
+
+		for( ContextVariablePlug::Iterator it( contextVariablesPlug() ); !it.done(); ++it )
+		{
+			const ContextVariablePlug *plug = it->get();
+			if( plug->enabledPlug()->getValue() )
+			{
+				plug->namePlug()->hash( h );
+				plug->quantizePlug()->hash( h );
+			}
+		}
+
+		if( timeOffsetPlug()->enabledPlug()->getValue() )
+		{
+			timeOffsetPlug()->namePlug()->hash( h );
+			timeOffsetPlug()->quantizePlug()->hash( h );
+		}
 	}
 	else if( output == prototypeChildNamesPlug() )
 	{
 		enginePlug()->hash( h );
 	}
+	else if( output == variationsPlug() )
+	{
+		// The sum of the variations across different engines depends on all the engines, but
+		// not their order.  We can create a cheap order-independent hash by summing the hashes
+		// all of the engines
+		std::atomic<uint64_t> h1Accum( 0 ), h2Accum( 0 );
+		std::function<bool(const GafferScene::ScenePlug *, const GafferScene::ScenePlug::ScenePath &)> functor =[this, &h1Accum, &h2Accum]( const GafferScene::ScenePlug *scene, const GafferScene::ScenePlug::ScenePath &path )
```

Comment:
I think we can just use `auto functor = `. As well as being more readable, I believe that means we're passing a lambda directly to `filteredParallelTraverse()` without an intervening `std::function()` wrapper.

Reply:
Yep, that's better.  I was just thrashing a bit after getting puzzled by the fact that filteredParallelTraverse won't take an inline lambda since it requires the functor to be non-const, and jumped on the first solution I found.

Reply:
Yeah, took me a little while to realise we didn't accept inline lambdas there as well. We should update SceneAlgo to allow that, but not in this PR.